viewing_party/party.py

```diff
@@ -1,23 +1,153 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating: 
+        return None
+    if title and genre and rating:
+        movie_dict = {}
+        movie_dict['title'] = title
+        movie_dict['genre'] = genre
+        movie_dict['rating'] = rating
+
+        return movie_dict
+
+    
+def add_to_watched(user_data, movie):
+    user_data['watched'].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie_dict in user_data['watchlist']:
+        if title == movie_dict['title']:
+            user_data['watched'].append(movie_dict)
+            user_data['watchlist'].remove(movie_dict)
+
+    return user_data
+
+    
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    total_rating = 0.0
+    avg_rating = 0.0
+    for movie_dict in user_data['watched']:
+        total_rating += movie_dict['rating']
+        if not len(user_data['watched']):
+            avg_rating = 0.0
+        else:
+            avg_rating = total_rating / len(user_data['watched'])

```

Comment:
If you move lines 43-44 outside of the loop, then you don't need the else statement. 

Do you mean for line 46 to run **inside** the for loop? Do we want to calculate the average value every time the loop iterates? What if you calculate total_rating within the for loop and then when the for loop is done running you could just return the value?

Also, note that you can drop the len() from user_data['watched'] because an empty list is falsy.

```python
def get_watched_avg_rating(user_data):
    if not user_data['watched']:
        return 0.0

    total_rating = 0.0
    for movie_dict in user_data['watched']:
        total_rating += movie_dict['rating']
    
    return total_rating / len(user_data['watched'])